### ISSUE TITLE
Qute asHtmlAttributes is including `it` as key="key" but should not

### ISSUE THREAD

**Issue by ia3andy** [quarkusio/quarkus#38280]
### Describe the bug

Qute asHtmlAttributes is including `it` as  key="key" and also as `"it"`. 

IMO we should:
- have a new method `includeKeyOnlyArgs(boolean includeFirst)`
- never have `"it"`

This way you really control the output of something like this `{#foo first arg2="aaa" arg3 arg4/}`

The boolean includeFirst determine if the first argument is meant to be used as a dynamic or as a `it`

**Comment by mkouba** [comment]
> The it is easy to filter out using skip("it") but I think it should always be ignored from the args list, it is an artificial arg (I don't see any use case where you would want the it in the args).

Agreed.

> sometime you want to skip them all because you only want to keep the key=value args => skipKeyOnlyArgs()

The problem with `skipKeyOnlyArgs()` is that in qute the single identifier is not actually a _key_ but a _value_ and the key/name is defaulted. So for example, `{#myTag foo}` becomes `{#myTag foo=foo}` and then `foo` is resolved from the current context and can be accessed in the tag using `{foo}`. 

In other words, `foo=foo` is not the same as `foo="foo"`, i.e. it does not mean that `{foo}` outputs `foo` in the tag. We deciced that in order to achieve the behavior of `foo="foo"` with a single identifier we will modify the behavior for string literals used as value-only arguments, i.e. `"foo"` will become `foo="foo"`. So that `{#myTag "foo"}` becomes `{#myTag foo="foo"}` and `{foo}` outputs `foo` in the tag template.

Now back to the `skipKeyOnlyArgs()` method - the name is not correct. We could add something like `skipValueOnlyStringLiteralArgs()` but I don't think it's very comprehensible. It might make sense to add something like `skipIdenticalKeyValue()`...

> sometime you want to only skip the first one because it's consumed as it => skipIt and you want to print the other one (for example readonly="readonly").

Given the fact that we always add a defaulted key even for the first unnamed arg and we will remove the `it` binding I'd say that it's always better to use the `_args.skip("foo")`.

**Comment by ia3andy** [comment]
> Given the fact that we always add a defaulted key even for the first unnamed arg and we will remove the `it` binding I'd say that it's always better to use the `_args.skip("foo")`.

You don't know the name which will be used, it could be anything, so there is no way to filter it:
ex: `{#gravatar email size='mm' readonly/}` one could also use `{#gravatar user.email size='mm' readonly/}` or `{#gravatar "email" size='mm' readonly/}`

**Comment by mkouba** [comment]
> > Given the fact that we always add a defaulted key even for the first unnamed arg and we will remove the `it` binding I'd say that it's always better to use the `_args.skip("foo")`.
> 
> You don't know the name which will be used, it could be anything, so there is no way to filter it

Yes, a tag "user" can pass the arguments in any order. But that means that `skipIt()` would also skip a "random" arg.

> ex: `{#gravatar email size='mm' readonly/}` one could also use `{#gravatar user.email size='mm' readonly/}` or `{#gravatar "foo@bar.io" size='mm' readonly/}`

- `{#gravatar email size='mm' readonly/}` -> `email` is `it` and also accessible using the defaulted key: `{email}`
- `{#gravatar user.email size='mm' readonly/}` -> `user.email` is `it` and the key is not defaulted because `user.email` is not a single identifier
- `{#gravatar "foo@bar.io" size='mm' readonly/}` -> `readonly` is `it` and also accessible using the defaulted key: `{readonly}`
